radiation/python/radsw/taumol_gt4py.py

```diff
@@ -206,6 +258,293 @@ def loadlookupdata(name):
+        if idxday:
+            for jb in range(nbandssw):
+                if laytrop:
+                    id0[0, 0, 0][jb] = ((jp - 1) * 5 + (jt - 1)) * nspa[0, 0][jb]
+                    id1[0, 0, 0][jb] = (jp * 5 + (jt1 - 1)) * nspa[0, 0][jb]
+                else:
+                    id0[0, 0, 0][jb] = ((jp - 13) * 5 + (jt - 1)) * nspb[0, 0][jb]
+                    id1[0, 0, 0][jb] = ((jp - 12) * 5 + (jt1 - 1)) * nspb[0, 0][jb]
+
+            for j in range(NG16):
+                sfluxzen[0, 0][ngs[0, 0][0] + j] = sfluxref01[0, 0, 0][
+                    j, 0, ibx[0, 0, 0][0]
+                ]
+            for j2 in range(NG20):
+                sfluxzen[0, 0][ngs[0, 0][4] + j2] = sfluxref01[0, 0, 0][
+                    j2, 0, ibx[0, 0, 0][4]
+                ]
+            for j3 in range(NG23):
+                sfluxzen[0, 0][ngs[0, 0][7] + j3] = sfluxref01[0, 0, 0][
+                    j3, 0, ibx[0, 0, 0][7]
+                ]
+            for j4 in range(NG25):
+                sfluxzen[0, 0][ngs[0, 0][9] + j4] = sfluxref01[0, 0, 0][
+                    j4, 0, ibx[0, 0, 0][9]
+                ]
+            for j5 in range(NG26):
+                sfluxzen[0, 0][ngs[0, 0][10] + j5] = sfluxref01[0, 0, 0][
+                    j5, 0, ibx[0, 0, 0][10]
+                ]
+            for j6 in range(NG29):
+                sfluxzen[0, 0][ngs[0, 0][13] + j6] = sfluxref01[0, 0, 0][
+                    j6, 0, ibx[0, 0, 0][13]
+                ]
+
+            for j7 in range(NG27):
+                sfluxzen[0, 0][ngs[0, 0][11] + j7] = (
+                    scalekur * sfluxref01[0, 0, 0][j7, 0, ibx[0, 0, 0][11]]
+                )
+
+    with computation(FORWARD), interval(0, -1):
+        if idxday:
+            if not laytrop:
+                # case default
+                cond = jp < layreffr[0, 0, 0][1] and jp[0, 0, 1] >= layreffr[0, 0, 0][1]
+
+                if cond:
+                    colm1 = colamt[0, 0, 1][ix1[0, 0, 0][1]]
+                    colm2 = colamt[0, 0, 1][ix2[0, 0, 0][1]]
+                elif sfluxzen[0, 0][ngs[0, 0][1]] == 0.0 and layind == 61:
+                    colm1 = colamt[0, 0, 1][ix1[0, 0, 0][1]]
+                    colm2 = colamt[0, 0, 1][ix2[0, 0, 0][1]]
+
+                if colm1 != 0.0:
+                    speccomb = colm1 + strrat[0, 0, 0][1] * colm2
+                    specmult = specwt[0, 0, 0][1] * min(oneminus, colm1 / speccomb)
+                    js = specmult
+                    fs = mod(specmult, 1.0)
+
+                    for jj in range(NG17):
+                        sfluxzen[0, 0][ngs[0, 0][1] + jj] = sfluxref02[0, 0, 0][
+                            jj, js, ibx[0, 0, 0][1]
+                        ] + fs * (
+                            sfluxref02[0, 0, 0][jj, js + 1, ibx[0, 0, 0][1]]
+                            - sfluxref02[0, 0, 0][jj, js, ibx[0, 0, 0][1]]
+                        )
+
+                colm1 = 0.0
+                colm2 = 0.0
```

Comment:
should this also happen before the first assignment of colm1/colm2? e.g. line 384. Is the check of colm1 != 0.0 also in the fortran code?

Reply:
I didn't because the input storages were made using `gt4py.storage.zeros`, so they will always be 0. The check is not in the fortran because this is implemented very differently in the fortran. Since `colm1` and `colm2` are 2D in the fortran, but 3D here, I only want the stencil to use the vertical layer where they are non-zero